### PR TITLE
Improve logging for indexer models and processors

### PR DESCRIPTION
**Pull request by dermanyang**
### Purpose
[Context for onboarding task](https://www.notion.so/aptoslabs/Sean-Onboarding-46a443876035440f8c577a31ef015974?pvs=4#fc60141e2c714ba7bc1d6ccb921f4621)

Additional metadata (`write_set_change_index` and/or `txn_version`) to logs for easier debugging. 
Namely, logs in 
* `coin_utils.rs`
*  `write_set_changes.rs`
*  `ans_processor.rs`
The rest of the diff is just making the appropriate variables available for the logging. 

### Testing
image
Ran `default_processor` (for wsc.unspecified logging),  `ans_processor` and `fungible_asset_processor` locally.

### CODE REVIEW

File: rust/processor/src/processors/ans_processor.rs [hunk 1]
```diff
@@ -582,6 +582,7 @@ fn parse_ans(
                             .unwrap_or_else(|e| {
                                 error!(
                                     error = ?e,
+                                    write_set_change_index = wsc_index,
```
Comment (rtso) [inline rust/processor/src/processors/ans_processor.rs:585]: Could we include the txn_version here?

File: rust/processor/src/db/common/models/default_models/write_set_changes.rs [hunk 2]
```diff
@@ -190,6 +191,10 @@ impl WriteSetChange {
             WriteSetChangeTypeEnum::WriteResource => "write_resource".to_string(),
             WriteSetChangeTypeEnum::WriteTableItem => "write_table_item".to_string(),
             WriteSetChangeTypeEnum::Unspecified => {
+                error!(
+                    "Encountered Unspecified WriteSetChange type. Transaction Version: {}",
```
Comment (rtso) [inline rust/processor/src/db/common/models/default_models/write_set_changes.rs:195]: Include wsc_index here?

File: rust/processor/src/db/common/models/default_models/write_set_changes.rs [hunk 3]
```diff
@@ -19,6 +19,7 @@ use aptos_protos::transaction::v1::{
 };
 use field_count::FieldCount;
 use serde::{Deserialize, Serialize};
+use tracing::log::error;
```
Comment (rtso) [inline rust/processor/src/db/common/models/default_models/write_set_changes.rs:22]: nit: tracing re-exports the macros at the top level
```suggestion
use tracing::error;
```

**Review by yuunlimm — APPROVED** [review]

**Review by rtso — APPROVED** [review]
Nice work!

**Pull request merged**